### PR TITLE
Theme improvements

### PR DESCRIPTION
**Pull request by Josmorsot**

### CODE REVIEW

**Review by alasarr — APPROVED** [review]

File: template/src/components/common/Header.js [hunk 1]
```diff
@@ -1,22 +1,42 @@
...
 
 const useStyles = makeStyles((theme) => ({
   navBar: {
     boxShadow: 'none',
   },
-  logo: {
-    height: '36px',
-    '& img': {
-      height: '100%',
+  divider: {
```
Comment (alasarr) [inline template/src/components/common/Header.js:20]: Cannot we add this to the theme? 

cc: @neokore

**Review by alasarr — CHANGES_REQUESTED** [review]

**Review by alasarr — CHANGES_REQUESTED** [review]
@neokore please review this PR since it's related with styles and theme.

**Comment by alasarr** [comment]
It breaks the styles of the header:
image

File: template/src/components/views/UserMenu.js [hunk 2]
```diff
@@ -12,6 +12,11 @@ const useStyles = makeStyles((theme) => ({
   menuButton: {
     marginRight: theme.spacing(2),
   },
+  avatar: {
+    cursor: 'pointer',
+    width: 36,
```
Comment (neokore) [inline template/src/components/views/UserMenu.js:17]: use spacing

File: template/src/lib/ui/stories/common/Dialog.stories.js [hunk 3]
```diff
@@ -83,4 +86,4 @@ export const Default = Template.bind({});
 Default.args = { content: <TextContent /> };
 
 export const Form = Template.bind({});
-Form.args = { content: <FormContent /> };
+Form.args = { content: <FormContent />, maxWidth: false, contentStyle: { width: 383 } };
```
Comment (neokore) [inline template/src/lib/ui/stories/common/Dialog.stories.js:89]: Why having to set styles manually?

File: template/src/lib/ui/stories/common/Palette.stories.js [hunk 4]
```diff
@@ -50,6 +50,24 @@ const ColorTemplate = ({ colorVariant, shade, ...args }) => {
           {colorDef.dark}
         </Typography>
       </Box>
+      {colorDef.relatedDark && (
+        <Box style={{ backgroundColor: colorDef.relatedDark, ...boxStyle }} {...args}>
```
Comment (neokore) [inline template/src/lib/ui/stories/common/Palette.stories.js:54]: What color is relatedDark or relatedLight?

File: template/src/lib/ui/stories/common/Palette.stories.js [hunk 5]
```diff
@@ -140,6 +158,44 @@ const BackgroundTemplate = (args) => {
...
+  return (
+    <Grid container>
+      <Box style={{ backgroundColor: colorDef.tooltip, ...boxStyle }} {...args}>
+        <Typography variant='caption' style={{ color: textColor }}>
+          Tooltip
+          <br />
+          {colorDef.tooltip}
+        </Typography>
+      </Box>
+      <Box style={{ backgroundColor: colorDef.snackbar, ...boxStyle }} {...args}>
```
Comment (neokore) [inline template/src/lib/ui/stories/common/Palette.stories.js:174]: I see lots of custom colors, do we really need them?

File: template/src/lib/ui/stories/common/Tooltip.stories.js [hunk 6]
```diff
@@ -119,13 +128,18 @@ const DataComponent = () => {
 };
 
 const TooltipDataTemplate = () => {
+  const DataTooltip = withStyles((theme) => ({
+    tooltip: {
+      padding: theme.spacing(1, 1.5),
+    },
+  }))(Tooltip);
```
Comment (neokore) [inline template/src/lib/ui/stories/common/Tooltip.stories.js:135]: The storybook should show what do you get with no custom styles. Maybe a bit of styling is ok if it is justified; but in this case I see maybe too mucho.

File: template/src/lib/ui/widgets/HistogramWidgetUI.js [hunk 7]
```diff
@@ -40,13 +40,25 @@ function __generateDefaultConfig(
...
   return {
     grid: {
-      left: 8,
-      top: 20,
-      right: 8,
-      bottom: 60,
+      left: 0,
+      top: 16,
+      right: 0,
+      bottom: 24,
```
Comment (neokore) [inline template/src/lib/ui/widgets/HistogramWidgetUI.js:46]: Use spacing

File: template/src/lib/ui/widgets/HistogramWidgetUI.js [hunk 8]
```diff
@@ -40,13 +40,25 @@ function __generateDefaultConfig(
...
+    },
+    axisPointer: {
+      lineStyle: {
+        color: theme.palette.charts.axisPointer,
+      },
     },
     tooltip: {
       trigger: 'axis',
+      padding: [4, 8],
```
Comment (neokore) [inline template/src/lib/ui/widgets/HistogramWidgetUI.js:55]: Can we use the tooltip from MaterialUI? If not, at least use spacing.

File: template/src/lib/ui/widgets/HistogramWidgetUI.js [hunk 9]
```diff
@@ -40,13 +40,25 @@ function __generateDefaultConfig(
...
+      textStyle: {
+        ...theme.typography.caption,
+        fontSize: 12,
+        lineHeight: 16,
```
Comment (neokore) [inline template/src/lib/ui/widgets/HistogramWidgetUI.js:59]: Is it a custom typography?

File: template/src/lib/ui/stories/common/Palette.stories.js (hunk 4 above)
Comment (Josmorsot) [reply template/src/lib/ui/stories/common/Palette.stories.js:54]: New palette colors: for primary, secondary, etc https://app.clubhouse.io/cartoteam/story/119457/qa-storybook-palette

File: template/src/lib/ui/stories/common/Palette.stories.js (hunk 5 above)
Comment (Josmorsot) [reply template/src/lib/ui/stories/common/Palette.stories.js:174]: see https://github.com/CartoDB/cra-template-carto/pull/86#discussion_r526024246

File: template/src/lib/ui/widgets/HistogramWidgetUI.js [hunk 10]
```diff
@@ -205,6 +223,7 @@ function HistogramWidgetUI(props) {
     tooltipFormatter,
     xAxisFormatter,
     yAxisFormatter,
+    height = 174,
```
Comment (neokore) [inline template/src/lib/ui/widgets/HistogramWidgetUI.js:226]: That's accurate! Maybe better with spacing?

File: template/src/lib/ui/widgets/HistogramWidgetUI.js [hunk 11]
```diff
@@ -299,9 +319,12 @@ function HistogramWidgetUI(props) {
 HistogramWidgetUI.propTypes = {
```
Comment (neokore) [inline template/src/lib/ui/widgets/HistogramWidgetUI.js:319]: Can we add defaultProps?

File: template/src/lib/ui/widgets/HistogramWidgetUI.js (hunk 9 above)
Comment (Josmorsot) [reply template/src/lib/ui/widgets/HistogramWidgetUI.js:59]: No, but Echarts doesn't support sizes as em, rem

**Review by neokore — CHANGES_REQUESTED** [review]
Some changes and questions to solve

File: template/src/lib/ui/widgets/HistogramWidgetUI.js (hunk 10 above)
Comment (Josmorsot) [reply template/src/lib/ui/widgets/HistogramWidgetUI.js:226]: it's the default size asked by design. I can change to spacing but I think that is not necessary

File: template/src/lib/ui/stories/common/Tooltip.stories.js (hunk 6 above)
Comment (Josmorsot) [reply template/src/lib/ui/stories/common/Tooltip.stories.js:135]: it's only for the specific case. Is not possible to apply two styles for the same tooltip. The only solution would be create a DataTooltip, but I think that it's not necessary. So, I think that we can remove this example or create the DataTooltip.
What do you think?

File: template/src/lib/ui/stories/common/Dialog.stories.js (hunk 3 above)
Comment (Josmorsot) [reply template/src/lib/ui/stories/common/Dialog.stories.js:89]: Asked by design  for this specific example

File: template/src/lib/ui/widgets/HistogramWidgetUI.js [hunk 12]
```diff
@@ -316,6 +316,17 @@ function HistogramWidgetUI(props) {
   );
 }
 
+HistogramWidgetUI.defaultProps = {
+  data: [],
```
Comment (neokore) [inline template/src/lib/ui/widgets/HistogramWidgetUI.js:320]: Default not needed for required props

File: template/src/lib/ui/widgets/HistogramWidgetUI.js (hunk 9 above)
Comment (neokore) [reply template/src/lib/ui/widgets/HistogramWidgetUI.js:59]: :+1:

File: template/src/lib/ui/stories/common/Dialog.stories.js (hunk 3 above)
Comment (neokore) [reply template/src/lib/ui/stories/common/Dialog.stories.js:89]: Better define widths on `paperWidthXs`,... See https://material-ui.com/api/dialog/

File: template/src/lib/ui/stories/common/Tooltip.stories.js (hunk 6 above)
Comment (Josmorsot) [reply template/src/lib/ui/stories/common/Tooltip.stories.js:135]: To fix

Comment (neokore) [reply template/src/lib/ui/stories/common/Tooltip.stories.js:135]: Better define custom padding on the `TooltipContent` component

File: template/src/lib/ui/widgets/HistogramWidgetUI.js (hunk 10 above)
Comment (neokore) [reply template/src/lib/ui/widgets/HistogramWidgetUI.js:226]: We should define default sizes respecting the grid, so I'd better go with 176 -> spacing(22).
Also we need to define a default widget size so they can be placed one next to each other and look good. Does it make sense for you?
/cc @javigil-geo @alasarr

File: template/src/lib/ui/stories/common/Palette.stories.js (hunk 4 above)
Comment (neokore) [reply template/src/lib/ui/stories/common/Palette.stories.js:54]: Ok. But maybe we should keep only colors from the MaterialUI theme, so the devs will know what to expect. If not, we should add them to the documentation and make it clear for them.

File: template/src/lib/ui/widgets/HistogramWidgetUI.js (hunk 10 above)
Comment (Josmorsot) [reply template/src/lib/ui/widgets/HistogramWidgetUI.js:226]: fixed

File: template/src/lib/ui/stories/common/Palette.stories.js (hunk 4 above)
Comment (Josmorsot) [reply template/src/lib/ui/stories/common/Palette.stories.js:54]: @alasarr @javigil-geo I think that we can go with palette for now and discuss it

Comment (alasarr) [reply template/src/lib/ui/stories/common/Palette.stories.js:54]: Agree with @neokore, but I'm going to merge and @Josmorsot will create a PR later removing the colors

**Review by alasarr — APPROVED** [review]

**Pull request merged**